apps/labs/posts/close-encounter-with-pandas-and-the-jedis-of-open-source.md

```diff
@@ -0,0 +1,247 @@
+consensus on how they should be tackled. Luckily for me, I didn't need to have midichlorians to contribute and, 
+most of all, I had mentors. Noa suggested that I tackle simpler issues to familiarize myself with the project and 
+learn the steps required to make contributions to pandas.
+
+Three issues and three pull requests later, I learned about building the Cython library every time I pulled new 
+changes from remote origin, how to always do pre-commit, how to write proper tests, how to submit issues and pull 
+requests, and how to document fixes.
+
+### Comments, tests, and a table
+
+> Being a developer is more than just typing code
+
+Having been involved in other internships, with the most recent being the [MLH Fellowship](https://fellowship.mlh.io) 
+and [Outreachy](https://www.outreachy.org/), I had to write a significant amount of code which was my 
+metric for measuring a developer at the time. This idea nursed a little bit of doubt in me as an intern 
+who was doing more reading and communicating rather than writing code. But luckily "*being a developer is 
+more than just typing code*", paraphrasing Malissa's comment when I made a pseudo-complaint about my 
+internship experience thus far (bearly a month in at the time).
+
+Since most people working on open-source projects are usually not paid to make contributions, there are 
+lots of features to be implemented, tests to be performed and documentation to write. Noa mentioned that 
+part of being a maintainer was doing actual maintenance and this involves things like "testing all pandas 
+operations with empty inputs and documenting the findings", "improving the documentation for some 
+operation or method" or "writing an internal documentation for a feature".
+
+There was a need to understand what was expected and having read the already cold discussion on the issue 
+to maintain consistency for panda's operation on empty inputs, It turned out though that there haven't 
+been many tests written on empty inputs and some of the results are a little unpredictable. For example, given
+
+```python
+    import pandas as pd
+
+    empty_df = pd.DataFrame({"a": [], "b": []})
+    gb = empty_df.groupby("a", group_keys=True)
+    group = getattr(gb, "b")
+    group.apply("sum") # Returns an empty series, which is understandable since the group is empty.
+    # But trying to get a sample of an empty group, returns an error
+    group.sample() # Returns ValueError: need at least one array to concatenate
+
+```
+Making this discovery was only possible because my mentors had advised me to create a list of pandas 
+operations and tested them with empty inputs. It was really brilliant because it gave a very clear picture 
+of the various inconsistencies in handling empty inputs and allowed for interest in the issue to be 
+rekindled. Here's a cutout from the table showing some GroupBy methods acting on empty dataframes.
+
+<table>
```

Comment:
Hi @Th3nn3ss! I see you already removed the style attributes. In general, I recommend against using HTML or inline styles. I recommend markup over HTML because it's easier for other people to read and edit as necessary. And I recommend against inline styles because it makes it harder for us to later change the look-and-feel of the website if components are inline styled.

But, if you really really need to use HTML tags and inline styles, then because this is a `.mdx` file you have to use [JSX](https://reactjs.org/docs/introducing-jsx.html), which is similar to HTML but not quite. For example, in order to use inline styles, you have to write it like so:

```jsx
<table style={{padding: "10px"}}>
```

Reply:
Thanks @gabalafou, I have tried using a markdown table instead, like the other mentors mentioned, but can't seem to get it to work. Code blocks with multiple lines don't tend to work inside table rows. Maybe am doing it wrong. Do you have a way I can overcome the issue with embedding code-blocks (with multiple lines) inside markdown table rows? I will really like to avoid any future issues with using HTML tags.

Reply:
I'll take a look at this right now.